### PR TITLE
nautilus: osd/OSDMap: Replace get_out_osds with get_out_existing_osds

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/39421

### CODE REVIEW

**Comment by tchaikov** [comment]
retest this please.

File: src/osd/OSDMap.h [hunk 1]
```diff
@@ -714,6 +714,7 @@ class OSDMap {
   void get_all_osds(set<int32_t>& ls) const;
   void get_up_osds(set<int32_t>& ls) const;
   void get_out_osds(set<int32_t>& ls) const;
+  void get_out_existing_osds(std::set<int32_t>& ls) const;
```
Comment (neha-ojha) [inline src/osd/OSDMap.h]: I'm not sure the resolution is correct here, we should not be using `std::` to be consistent.

Comment (badone) [reply src/osd/OSDMap.h]: But it's a header so *all* references to std:: objects should be fully qualified. There is no "using namespace std" in the file so the lines above it are incorrect and only work by good fortune :) If anything, all the other lines need to be fully qualified but I don't believe that is in the scope of the backport and would tend to obfuscate git history anyway so probably not worth it at this time.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2019-05-23-2026-nautilus](https://trello.com/c/ofkznvrv/618-wip-yuri3-testing-2019-05-23-2026-nautilus)

**Comment by badone** [comment]
jenkins retest this please

**Comment by tchaikov** [comment]
jenkins test docs please

**Pull request merged**